### PR TITLE
Use fantasticon to generate web fonts

### LINKED ISSUE #86: Provide font-family format
The "_You can also use the SVG within your CSS_" use case doesn't allow for color customization without touching the SVG `fill` value. We should be able to dynamically load the untouched source SVGs and customize them via css only (like when you import Bootstrap without modifying it and properly override variables), which would be possible via a font-family approach (allowing the use of Bootstrap color variables for proper icons theming). Is this in the pipeline?

### PR DESCRIPTION
**Pull request by mdo**
Replaces #287 with a new dependency, fantasticon. Fixes #86.

### CODE REVIEW

File: docs/assets/scss/docs.scss [hunk 1]
```diff
@@ -140,3 +140,29 @@
   padding-bottom: 4.5rem !important;
 }
 // stylelint-enable declaration-no-important
+
+// stylelint-disable
```
Comment (XhmikosR) [inline docs/assets/scss/docs.scss:144]: Better be more explicit here?

File: .fantasticonrc.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  pathOptions: {
+    json: './font/bootstrap-icons.json',
+    css: './font/bootstrap-icons.css',
+    html: './font/index.html',
+    ttf: './font/fonts/bootstrap-icons.ttf',
+    woff: './font/fonts/bootstrap-icons.woff',
+    woff2: './font/fonts/bootstrap-icons.woff2',
+    eot: './font/fonts/bootstrap-icons.eot'
+  }
+};
```
Comment (XhmikosR) [inline .fantasticonrc.js:30]: EOF newline

File: src/html.hbs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  {{# each assets }}
+    <div class="icon">
+      <{{ ../tag }} class="{{ ../prefix }} {{ ../prefix }}-{{ @key }}"></{{ ../tag }}>
+      <div class="label">{{ @key }}</div>
+    </div>
+  {{/ each }}
+  </div>
+
+</body>
+</html>
```
Comment (XhmikosR) [inline src/html.hbs:53]: EOF newline

File: package.json [hunk 4]
```diff
@@ -61,7 +62,8 @@
   },
   "files": [
     "icons/*.svg",
-    "bootstrap-icons.svg"
+    "bootstrap-icons.svg",
+    "fonts"
```
Comment (XhmikosR) [inline package.json:66]: Better be more explicit here like this:

```suggestion
    "fonts/*.{eot,ttf,woff,woff2}"
```

**Review by XhmikosR — CHANGES_REQUESTED** [review]
Also,

* the hbs files seem unused
* package.json is missing the new devDependency
* the script should be added to another script (or we run it separately) so that's tested on CI

File: package.json (hunk 4 above)
Comment (mdo) [reply package.json:66]: I want the whole fonts directory for now. Too weird?

File: docs/assets/scss/docs.scss (hunk 1 above)
Comment (mdo) [reply docs/assets/scss/docs.scss]: Done, but it's a mouthful haha.

**Comment by mdo** [comment]
> * the hbs files seem unused

`.hbs` files are indeed used—they help generate the CSS and HTML in the `font` directory. They're specified in `.fantasticonrc.js`.

**Comment by mdo** [comment]
> * package.json is missing the new devDependency
> * the script should be added to another script (or we run it separately) so that's tested on CI

Done and done.

- Hooked into the `icons` script
- Added devDependency
- Renamed the `font` script to `icons-font` for consistency
- Added the `font` dir to the zip script

File: package.json (hunk 4 above)
Comment (XhmikosR) [reply package.json:67]: Yup. Because unneeded files might be included in the npm package, even files not tracked by git.

**Comment by XhmikosR** [comment]
Just a question: do you really want to add eot and ttf fonts? woff and woff2 should be more than enough. Heck, you could even just generate woff2 if you don't care about IE :P

* https://caniuse.com/?search=woff
* https://caniuse.com/?search=woff2

File: package.json (hunk 4 above)
Comment (XhmikosR) [reply package.json:67]: Actually, now that I see the folder contents, ignore my comment. It's just too many filetypes to specify.

File: src/html.hbs [hunk 5]
```diff
@@ -0,0 +1,53 @@
+<!doctype html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
```
Comment (XhmikosR) [inline src/html.hbs:4]: Maybe lowercase this?

File: src/html.hbs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  <title>{{ name }}</title>
+
+  <style>
+    .icons {
+      display: grid;
+      max-width: 100%;
+      grid-template-columns: repeat(auto-fit, minmax(100px, 1fr) );
```
Comment (XhmikosR) [inline src/html.hbs:11]: Shouldn't this be prefixed?

Comment (mdo) [reply src/html.hbs:11]: I don't think so? https://deploy-preview-31813--twbs-bootstrap.netlify.app/docs/5.0/layout/css-grid/ doesn't show any prefixing...

Comment (XhmikosR) [reply src/html.hbs:11]: Yeah, it needs to be manually made since that's not handled by any tools: it's hardcoded.

Comment (XhmikosR) [reply src/html.hbs:11]: NVM, sorry, I just re-read what you wrote. Indeed, it's fine as is :)

**Comment by mdo** [comment]
Dropping EOT and TTF for now since we can always add back later, but removing is always more difficult.

Good to go after this I think!

File: docs/assets/scss/docs.scss (hunk 1 above)
Comment (XhmikosR) [reply docs/assets/scss/docs.scss]: Indeed, better revert this one, sorry!

**Comment by XhmikosR** [comment]
Minus my last comment, I think this looks good :)

**Comment by XhmikosR** [comment]
I wonder though, is fonts/index.html supposed to be served? It's not using relative paths to the CSS AFAICT.

**Comment by mdo** [comment]
> I wonder though, is fonts/index.html supposed to be served? It's not using relative paths to the CSS AFAICT.

Yeah, I want that served. It's used locally for previewing, and eventually I'll have it up to date and part of the full nav.

**Comment by XhmikosR** [comment]
Actually, I just saw you hooked the font directory as font so it's already available indeed: https://deploy-preview-515--bootstrap-icons.netlify.app/font/ and included in sitemap.

**Review by XhmikosR — APPROVED** [review]

**Comment by TechQuery** [comment]
@mdo @XhmikosR There is no **CSS & Web Font** files in the [ZIP bundle][1] & [NPM package][2]...

[1]: https://github.com/twbs/icons/releases/download/v1.2.0/bootstrap-icons-1.2.0.zip
[2]: https://www.jsdelivr.com/package/npm/bootstrap-icons

**Comment by XhmikosR** [comment]
The zip file does have the font for me. For the npm package see #538 .

**Pull request merged**